spec/controllers/catalog_controller_spec.rb

```diff
@@ -1,46 +1,97 @@
 require 'rails_helper'
-
 RSpec.describe CatalogController, type: :controller do
-  describe 'POST #add_preserved_object' do
+  describe 'POST #create' do
+    let(:size) { 2342 }
+    let(:ver) { 3 }
+    let(:endpoint_name) { 'fixture_sr1' }
+    let(:druid) { 'bj102hs9687' }
+
     context 'with valid params' do
-      it 'saves the object with the expected field values' do
-        allow(Stanford::StorageServices).to receive(:current_version).and_return(6)
-        allow(subject).to receive(:moab_size).and_return(555)
-        id = 'ab123cd45678'
+      let(:pres_copy) { PreservedCopy.first }
+      let(:pres_obj) { PreservedObject.first }
+
+      before do
+        post :create, params: { druid: druid, incoming_version: ver, incoming_size: size, endpoint_name: endpoint_name }
+      end
+
+      it 'saves PreservedObject and PreservedCopy in db' do
+        expect(PreservedObject).to exist
+        expect(PreservedCopy).to exist
+      end
+      it 'PreservedCopy and PreservedObject have correct attributes' do
+        expect(pres_copy.endpoint.endpoint_name).to eq endpoint_name
+        expect(pres_copy.version).to eq ver
+        expect(pres_copy.size).to eq size
+        expect(pres_obj.druid).to eq druid
+      end
 
-        allow(PreservedObject).to receive(:create!)
-        post :add_preserved_object, params: { id: id }
-        expect(PreservedObject).to have_received(:create!).with(druid: id, size: 555, current_version: 6)
+      it 'response contains create_new_object code ' do
+        exp_msg = [{ AuditResults::CREATED_NEW_OBJECT => "added object to db as it did not exist" }]
+        post_response = response.body
+        expect(post_response).to eq exp_msg.to_json
```

Comment:
i'd be inclined to do `expect(response.body)` directly without the temp var, but i don't have strong feelings and if you fine this style more readable, i'm totally fine with that.

Reply:
i changed it in the other places too